File: lib/models/product.dart
class Product {
  final int id;
  final String name;
  final double caloriesPer100g;
  final double proteinPer100g;
  final double carbsPer100g;
  final double fatPer100g;
  final double defaultGrams;
  final String? category;
  final bool isCustom;
  final DateTime createdAt;

  Product({
    required this.id,
    required this.name,
    required this.caloriesPer100g,
    required this.proteinPer100g,
    required this.carbsPer100g,
    required this.fatPer100g,
    this.defaultGrams = 100,
    this.category,
    this.isCustom = false,
    required this.createdAt,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] as int,
      name: json['name'] as String,
      caloriesPer100g: (json['caloriesPer100g'] as num).toDouble(),
      proteinPer100g: (json['proteinPer100g'] as num).toDouble(),
      carbsPer100g: (json['carbsPer100g'] as num).toDouble(),
      fatPer100g: (json['fatPer100g'] as num).toDouble(),
      defaultGrams: (json['defaultGrams'] as num?)?.toDouble() ?? 100,
      category: json['category'] as String?,
      isCustom: json['isCustom'] as bool? ?? false,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'caloriesPer100g': caloriesPer100g,
      'proteinPer100g': proteinPer100g,
      'carbsPer100g': carbsPer100g,
      'fatPer100g': fatPer100g,
      'defaultGrams': defaultGrams,
      'category': category,
      'isCustom': isCustom,
      'createdAt': createdAt.toIso8601String(),
    };
  }

  Product copyWith({
    int? id,
    String? name,
    double? caloriesPer100g,
    double? proteinPer100g,
    double? carbsPer100g,
    double? fatPer100g,
    double? defaultGrams,
    String? category,
    bool? isCustom,
    DateTime? createdAt,
  }) {
    return Product(
      id: id ?? this.id,
      name: name ?? this.name,
      caloriesPer100g: caloriesPer100g ?? this.caloriesPer100g,
      proteinPer100g: proteinPer100g ?? this.proteinPer100g,
      carbsPer100g: carbsPer100g ?? this.carbsPer100g,
      fatPer100g: fatPer100g ?? this.fatPer100g,
      defaultGrams: defaultGrams ?? this.defaultGrams,
      category: category ?? this.category,
      isCustom: isCustom ?? this.isCustom,
      createdAt: createdAt ?? this.createdAt,
    );
  }
}

File: FoodLogic.Core/Models/Product.cs
using System.Text.Json.Serialization;

namespace FoodLogic.Core.Models;

public class Product
{
    [JsonPropertyName("id")]
    public long Id { get; set; }

    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("caloriesPer100g")]
    public double CaloriesPer100g { get; set; }

    [JsonPropertyName("proteinPer100g")]
    public double ProteinPer100g { get; set; }

    [JsonPropertyName("carbsPer100g")]
    public double CarbsPer100g { get; set; }

    [JsonPropertyName("fatPer100g")]
    public double FatPer100g { get; set; }

    [JsonPropertyName("defaultGrams")]
    public double DefaultGrams { get; set; } = 100;

    [JsonPropertyName("category")]
    public string? Category { get; set; }

    [JsonPropertyName("isCustom")]
    public bool IsCustom { get; set; }

    [JsonPropertyName("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
